### PR TITLE
Only compile FieldPropsDataHandle if MPI was found

### PR DESCRIPTION
**Pull request by blattms**
Fixes the fallout form #2406 (breaking serial build)

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this serial please

File: opm/simulators/utils/FieldPropsDataHandle.hpp [hunk 1]
```diff
@@ -199,3 +200,4 @@ class FieldPropsDataHandle
 
 } // end namespace Opm
 #endif // FIELDPROPS_DATAHANDLE_HPP
+#endif // HAVE_MPI
```
Comment (akva2) [inline opm/simulators/utils/FieldPropsDataHandle.hpp]: wrong order

Comment (blattms) [reply opm/simulators/utils/FieldPropsDataHandle.hpp]: Fixed.

**Comment by blattms** [comment]
jenkins build this serial please

**Comment by akva2** [comment]
unless github is being weird, it's still the same. the mpi ifdef closes first.

**Comment by blattms** [comment]
I missed the -a option

**Comment by blattms** [comment]
jenkins build this serial please

**Comment by blattms** [comment]
shameless selfemerge to restore build.

**Pull request merged**